Src/DependencyCollector/Shared/Implementation/HttpHandlerDiagnosticListener.cs

```
@@ -0,0 +1,361 @@
+
+        // Fields for reflection
+        private static FieldInfo connectionGroupListField;
+        private static Type connectionGroupType;
+        private static FieldInfo connectionListField;
+        private static Type connectionType;
+        private static FieldInfo writeListField;
+        private static Func<HttpWebRequest, HttpWebResponse> httpResponseAccessor;
+
+        // Fields for controlling initialization of the HttpHandlerDiagnosticListener singleton
+        private object initializationLock = new object();
+        private bool initialized = false;
+#endregion
+
+        /// <summary>
+        /// Prevents a default instance of the <see cref="HttpHandlerDiagnosticListener" /> class from being created.
+        /// Make sure every caller has to create this object via the Initialize method.
+        /// </summary>
+        private HttpHandlerDiagnosticListener() : base(DiagnosticListenerName)
+        {
+        }
+
+        /// <summary>
+        /// Overriding base class implementation just to give us a chance to initialize.  
+        /// </summary>
+        public override IDisposable Subscribe(IObserver<KeyValuePair<string, object>> observer, Predicate<string> isEnabled)
+        {
+            this.Initialize();
+            return base.Subscribe(observer, isEnabled);
+        }
+
+        /// <summary>
+        /// Overriding base class implementation just to give us a chance to initialize.  
+        /// </summary>
+        public override IDisposable Subscribe(IObserver<KeyValuePair<string, object>> observer, Func<string, object, object, bool> isEnabled)
+        {
+            this.Initialize();
+            return base.Subscribe(observer, isEnabled);
+        }
+
+#region private methods
+
+        private static void PrepareReflectionObjects()
+        {
+            // At any point, if the operation failed, it should just throw, including NullReferenceException.
```

If it's not removed from 2.4 by stable milestone, I would recommend to implement `TryPrepare..()` in 2.4 context rather than wait for CoreFX, we would like to minimize # of the exceptions that are just thrown/swallowed especially on the "happy path".